oops! 

It's my mistake. I'm fixing the problem and will definitely run the testing process again before submitting.

```
FAILED: src/test/rgw/CMakeFiles/unittest_rgw_cors.dir/test_rgw_cors.cc.o 
/usr/local/bin/sccache /usr/bin/clang++-19 -DBOOST_ASIO_DISABLE_THREAD_KEYWORD_EXTENSION -DBOOST_ASIO_NO_TS_EXECUTORS -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_URL_DYN_LINK -DBOOST_URL_NO_LIB -DCLS_CLIENT_HIDE_IOCTX -DHAVE_CONFIG_H -DXXH_NO_INLINE_HINTS=1 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_REENTRANT -D_THREAD_SAFE -D__CEPH__ -D__STDC_FORMAT_MACROS -D__linux__ -I/ceph/src/dmclock/support/src -I/ceph/src/rgw -I/ceph/src/rgw/driver/rados -I/ceph/src/rgw/services -I/usr/include/lua5.3 -I/ceph/src/BLAKE3/c -isystem /opt/ceph/include -isystem /ceph/build/src/include -isystem /ceph/src -isystem /ceph/build/include -isystem /ceph/src/jaegertracing/opentelemetry-cpp/api/include -isystem /ceph/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/include -isystem /ceph/src/jaegertracing/opentelemetry-cpp/ext/include -isystem /ceph/src/jaegertracing/opentelemetry-cpp/sdk/include -isystem /ceph/src/xxHash -isystem /ceph/src/fmt/include -isystem /ceph/src/googletest/googlemock/include -isystem /ceph/src/googletest/googlemock -isystem /ceph/src/googletest/googletest/include -isystem /ceph/src/googletest/googletest -isystem /ceph/src/s3select/rapidjson/include -g -Werror -fPIE -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -Wall -fno-strict-aliasing -fsigned-char -Wtype-limits -Wignored-qualifiers -Wpointer-arith -Werror=format-security -Winit-self -Wno-unknown-pragmas -Wnon-virtual-dtor -Wno-ignored-qualifiers -Wsign-compare -ftemplate-depth-1024 -DBOOST_ALLOW_DEPRECATED_HEADERS -Wpessimizing-move -Wredundant-move -Wno-inconsistent-missing-override -Wno-mismatched-tags -Wno-unused-private-field -Wno-address-of-packed-member -Wno-unknown-warning-option -Wno-unused-function -Wno-unused-local-typedef -Wno-varargs -Wno-gnu-designator -Wno-missing-braces -Wno-parentheses -Wno-deprecated-register -Wno-vla-cxx-extension -DCEPH_DEBUG_MUTEX -D_GLIBCXX_ASSERTIONS -fdiagnostics-color=auto -Werror=vla -std=c++2b -MD -MT src/test/rgw/CMakeFiles/unittest_rgw_cors.dir/test_rgw_cors.cc.o -MF src/test/rgw/CMakeFiles/unittest_rgw_cors.dir/test_rgw_cors.cc.o.d -o src/test/rgw/CMakeFiles/unittest_rgw_cors.dir/test_rgw_cors.cc.o -c /ceph/src/test/rgw/test_rgw_cors.cc
In file included from /ceph/src/test/rgw/test_rgw_cors.cc:6:
/ceph/src/rgw/rgw_cors.h:172:3: error: use of undeclared identifier 'for_each_substr'
  172 |   for_each_substr(allowed_methods, ";,= \t", apply_flag);
      |   ^
1 error generated.
[2663/2986] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/rgw_cksum.cc.o
[2664/2986] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/rgw_bucket_layout.cc.o
[2665/2986] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/rgw_asio_thread.cc.o
[2666/2986] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/rgw_ldap.cc.o
[2667/2986] Building CXX object src/test/rgw/CMakeFiles/unittest_rgw_sync_trace.dir/test_rgw_sync_trace.cc.o
[2668/2986] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/rgw_cors.cc.o
[2669/2986] Generating ../../../lib/cython_modules/lib.3/rbd.cpython-310-x86_64-linux-gnu.so
Compiling rbd_processed.pyx because it changed.
```

--- a/src/rgw/rgw_cors.h
+++ b/src/rgw/rgw_cors.h
@@ -18,6 +18,7 @@
 #include <map>
 #include <string>
 #include <include/types.h>
+#include "include/str_list.h"
 
 #define RGW_CORS_GET    0x1
 #define RGW_CORS_PUT    0x2
@@ -169,7 +170,7 @@ static inline uint8_t get_multi_cors_method_flags(const char *req_meth) {
     else if (method == "HEAD") flags |= RGW_CORS_HEAD;
     else if (method == "COPY") flags |= RGW_CORS_COPY;
   };
-  for_each_substr(allowed_methods, ";,= \t", apply_flag);
+  ceph::for_each_substr(allowed_methods, ";,= \t", apply_flag);
 
   return flags;
 }

--- a/src/test/rgw/CMakeLists.txt
+++ b/src/test/rgw/CMakeLists.txt
@@ -103,7 +103,9 @@ add_ceph_unittest(unittest_rgw_bencode)
 target_link_libraries(unittest_rgw_bencode ${rgw_libs})
 
 # unittest_rgw_cors
-add_executable(unittest_rgw_cors test_rgw_cors.cc)
+add_executable(unittest_rgw_cors
+  test_rgw_cors.cc
+  $<TARGET_OBJECTS:unit-main>)
 add_ceph_unittest(unittest_rgw_cors)
 target_link_libraries(unittest_rgw_cors ${rgw_libs})
 

